pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ExclusiveProducerTest.java

```diff
@@ -118,7 +120,22 @@ private void simpleTest(String topic) throws Exception {
                 .topic(topic)
                 .accessMode(ProducerAccessMode.Exclusive)
                 .create();
-        p2.close();
+
+        Producer<String> p3 = pulsarClient.newProducer(Schema.STRING)
+                .topic(topic)
+                .accessMode(ProducerAccessMode.ExclusiveWithFencing)
+                .create();
+
+        try {
+            p2.send("test");
+            fail("Should have failed");
+        } catch (ProducerFencedException expected) {
+        }
+
+        // this should work
+        p3.send("test");
+        p3.close();
```

Comment:
I'd also add a test to verify that any queued up producer is also immediately fenced. eg: mixing up `WaitForExclusive` & `ExclusiveWithFencing`